components/script/dom/document.rs

```diff
@@ -2154,7 +2153,7 @@ impl Document {
                                         HasBrowsingContext::No,
                                         None,
                                         // https://github.com/whatwg/html/issues/2109
-                                        Origin::opaque_identifier(),
+                                        self.origin.clone(),
```

Comment:
Is this the only remaining issue? I can revert it if we want to keep the old behaviour.